cpp/src/arrow/testing/util.h

```diff
@@ -89,8 +90,8 @@ inline Status CopyBufferFromVector(const std::vector<T>& values, MemoryPool* poo
 // Sets approximately pct_null of the first n bytes in null_bytes to zero
 // and the rest to non-zero (true) values.
 ARROW_EXPORT void random_null_bytes(int64_t n, double pct_null, uint8_t* null_bytes);
-ARROW_EXPORT void random_is_valid(int64_t n, double pct_null,
-                                  std::vector<bool>* is_valid);
+ARROW_EXPORT void random_is_valid(int64_t n, double pct_null, std::vector<bool>* is_valid,
+                                  int random_seed = 0);
```

Comment:
Probably fine: elsewhere `uint32_t` is used for seeds, while `random_seed()` returns `uint64_t`.

Reply:
This was a quick fix for compatibility for one of the Parquet test generation functions, we can always change to `uint32_t` everywhere